pkg/validator/cluster.go

```diff
@@ -379,11 +380,14 @@ func validateGeneralClusterChanges(
 		}
 
 		// App version should be different from previous one
-		if prevCrApp.Spec.Version == crApp.Spec.Version {
+		prevVer := semver.New(prevCrApp.Spec.Version)
+		newVer := semver.New(crApp.Spec.Version)
+
```

Comment:
At this point, prevVer and/or newVer can be nil if the version string is not understood by semver. In those cases we should also reject the upgrade.

Reply:
Actually, my mistake. In those cases, `New` will panic, which is bad for us.

Try using NewVersion (instead of New) as in https://github.com/coreos/go-semver/blob/main/example.go

Reply:
Changed